components/raftstore/src/store/fsm/store.rs

```diff
@@ -2744,13 +2761,31 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
                 .swap(0, Ordering::Relaxed),
         );
 
-        stats.set_is_busy(
-            self.ctx
-                .global_stat
-                .stat
-                .is_busy
-                .swap(false, Ordering::Relaxed),
-        );
+        let store_is_busy = self
+            .ctx
+            .global_stat
+            .stat
+            .is_busy
+            .swap(false, Ordering::Relaxed);
+        let pending_on_apply = {
+            (time::get_time().sec as u32).saturating_sub(start_time)
+                <= STORE_RECOVERY_DURATION.as_secs() as u32
+        };
+        let is_busy = if pending_on_apply {
+            // If the store is busy in handling applying logs when starting, it should not
+            // be treated as a normal store for balance. Only when the store is
+            // almost idle (no more pending regions on applying logs), it can be
+            // regarded as the candidate for balancing leaders.
+            let target_count = std::cmp::max(
+                1,
+                self.ctx.cfg.min_apply_ready_region_percent * (stats.get_region_count() as u64)
```

Comment:
In that case, the configuration `min_apply_ready_region_percent` can be set `== 100` to make the store `is_busy == true` until all regions finish applying logs.

Reply:
Region count varies from cluster to cluster, then it would often need to adjust the config. Why not use a absolute number instead of percent

Reply:
Not good to me.

Using absolute number still has limitations. There exists one false-positive case which will make the `is_busy == true` incorrectly.

> PS: Assuming there are "10k" peers in this node, and the count limitation is set to 20. And we count the peer, not ready on finishing applying logs, with `uncompleted_on_apply_logs`.
That is, if the count of `uncompleted_on_apply_logs` < 20, the store will be marked with `store_is_busy == false`. Else, it will be `true`.

In a corner case, if the collection of `uncompleted_on_apply_logs` peers is not refreshed as expected — let's say, only "10 peers are pending on applying logs" in the first tick of `pd-store-heartbeat`— the check will pass since the count of `uncompleted_on_apply_logs` is 10, which is less than the limitation of `20`. Consequently, in this heartbeat tick, `store_is_busy` will be set to `false` as the condition meets the setting.

However, in the next tick, the collection of `uncompleted_on_apply_logs` is refreshed and updated with 1000. Now, it does not meet the condition, and `store_is_busy` is reported as `true`.

It's unexpected.

Reply:
As discussed offline, the `min_apply_ready_region_percent` has been replaced with `min_pending_apply_region_count`, which represents the minimal requirement for the count of uncompleted peers on applying logs.